### DISCUSSION TITLE
BFF Api 如何返回错误

### DISCUSSION THREAD

**Discussion by xmsz** [web-infra-dev/modern.js#3941] [Q&A] [answered]
第一个问题：如何返回错误
比如，400
我用`throw new HttpError(400, '123123')`, 但状态是变成了400，messge还是`Bad Request`

答案：
```
    const ctx = useContext(); 
    ctx.throw(new HttpError(400, "message"));
```

第二个问题：api出现错误，如何统一拦截返回错误
因为现在出错，返回的是`Internal Server Error` 不知道是什么

答案：
`_app.ts`内进行统一拦截 例如 await next() throw

**Comment by yimingjfe** [comment, marked as answer]
这里根据使用的运行时框架，按照框架的官方推荐的方式处理即可，比如你使用 Koa.js，就可以通过 `ctx.throw` 抛出错误，监听 onerror 事件处理错误
